bxserum/connections/http.go

```
@@ -1 +1,40 @@
 package connections
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// HTTP response for GET request
+func HTTPGetResponse[T any](client *http.Client, url string) (*T, error) {
+	if client == nil {
```

It'd be probably be good to make two functions: `HTTPGet` and `HTTPGetWithClient` and give the caller the default http client

---

I think usually you also return an `error` for this kind of `nil` pointer problem –– you'll usually just let the code panic so the developer who made this mistake gets alerted since it's not really part of normal operation.

---

For the second point, should we only panic when it is the developer's fault?

I.e. Should we return an error if the HTTP response from the client is nil? Yes, I think?

---

yes, panic is usually for when it's the developer fault or some irrecoverable invariant has been broken (i.e. user error in startup arguments). In your example, a server returning an empty response is usually not the developer or user's fault.